
<div style="border-radius:10px;
            border : #015a2c solid;
            background-color:#D7D9DB;
           font-size:110%;
           letter-spacing:0.5px;
            text-align: center">
​
<center><h1 style="padding: 25px 0px; color:#015a4c; font-weight: bold; font-family: Cursive">
Amit Project for Data Science Module <br><br>💡💡💡💡💡 <br><br> AI Online 74  </h1></center>
<center><h3 style="padding-bottom: 25px; color:#025b2c; font-weight: bold; font-style:italic; font-family: Cursive">
Using Project: Trip data Ford go bike  </h3></center>     
​
</div>

# Part I - (Dataset Exploration for Ford GoBike)

### worked by (Mohamed Abdeen Mostafa Hedar)

## Introduction

.

> **Let's introduce our dataset:**


======================================

.

**This data set information was produced from `Ford GoBike` System** the information in it talked about individual rides that were made in a bike-sharing system for covering San Francisco Bay area.** 

.




> ## And here are the main exploring questions 



   **==================================================**



- What is the structure of your dataset?


- What are the main interesting features in the dataset?


- What features in the dataset supported my investigation for these interesting features?




.

> ### Also here are the investigative questions classified into three types



   **================================================**
   
   


###  <a href='#011'> Univariate Exploration</a> 

- <a href='#1'> 1 ) . What is the distribution of `User type`?</a>  
- <a href='#2'> 2 ) . What is the distribution for the `bike share users on all Trips`?</a>  
- <a href='#3'> 3 ) . What is the distribution for users' `ages`?</a>  
- <a href='#4'> 4 ) . Do the users' ages have `outlier` values?</a>
- <a href='#5'> 5 ) . What is the distribution for the trip duration in `minutes`? </a>
- <a href='#6'> 6) . What is the distribution for `member_gender`? </a>
- <a href='#7'> 7) . What is the distribution for the trip duration in `Hours`?  </a>




##  <a href='#111'> Bivariate Exploration</a> 


- <a href='#8'> 8 ) . What the `average duration` of trip for the categories of `user type`? </a>
- <a href='#9'> 9 ) . How long is the duration of the trip for `each gender` in `minutes`? </a>
- <a href='#10'> 10 ) . How long is the duration of the trip for each `user_type` in `minutes`?  </a>
- <a href='#11'> 11 ) . How the user distribution depending on their `Age` and `user_type` ?</a>
- <a href='#12'>12 ). How the user distribution depending on their `Age` and `member_gender` ?  </a>


##  <a href='#1111'> Multivariate Exploration</a> 


- <a href='#13'> 13 ) . What is the Correlation between `each` variable in the dataset?</a>  
- <a href='#14'> 14 ) . What is the Correlation between `specific numeric variables` in the dataset?</a>  
- <a href='#15'> 15 ) . What is the  Correlation between `data_stat` variables in the dataset?</a>  


##  <a href='#1100'> Conclusions </a> 





# <p style="border-radius:15px;border-style:solid;border-color:lightblue;border-width:3px;background-color:lightblue;padding: 20px;font-family:Verdana;color:#D81F26;font-size:30px;text-align: center;">Import Libraries</p>

In [2]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import os
import matplotlib.pyplot as plt
import seaborn as sb
import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

In [3]:
# Load my dataset using pandas Dataframe
data = pd.read_csv(r"C:\DEPI\machine_learning\sessions\week_6\session_11,12\data\fordgobike-tripdataFor201902.csv")
df = data.copy()

In [4]:
# here we make overview in high-level for data shape and composition
df.head()

duration_sec start_time end_time  start_station_id  \
0         52185    32:10.1  01:56.0              21.0   
1         42521    53:21.8  42:03.1              23.0   
2         61854    13:13.2  24:08.1              86.0   
3         36490    54:26.0  02:36.8             375.0   
4          1585    54:18.5  20:44.1               7.0   

                                 start_station_name  start_station_latitude  \
0  Montgomery St BART Station (Market St at 2nd St)               37.789625   
1                     The Embarcadero at Steuart St               37.791464   
2                           Market St at Dolores St               37.769305   
3                           Grove St at Masonic Ave               37.774836   
4                               Frank H Ogawa Plaza               37.804562   

   start_station_longitude  end_station_id  \
0              -122.400811            13.0   
1              -122.391034            81.0   
2              -122.426826             3.0   
3              -122.446546            70.0   
4              -122.271738           222.0   

                               end_station_name  end_station_latitude  \
0                Commercial St at Montgomery St             37.794231   
1                            Berry St at 4th St             37.775880   
2  Powell St BART Station (Market St at 4th St)             37.786375   
3                        Central Ave at Fell St             37.773311   
4                         10th Ave at E 15th St             37.792714   

   end_station_longitude  bike_id   user_type  member_birth_year  \
0            -122.402923     4902    Customer             1984.0   
1            -122.393170     2535    Customer                NaN   
2            -122.404904     5905    Customer             1972.0   
3            -122.444293     6638  Subscriber             1989.0   
4            -122.248780     4898  Subscriber             1974.0   

  member_gender bike_share_for_all_trip  
0          Male                      No  
1           NaN                      No  
2          Male                      No  
3         Other                      No  
4          Male                     Yes

In [5]:
## check each var  datatype 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 183416 entries, 0 to 183415
Data columns (total 16 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   duration_sec             183416 non-null  int64  
 1   start_time               183416 non-null  object 
 2   end_time                 183416 non-null  object 
 3   start_station_id         183219 non-null  float64
 4   start_station_name       183219 non-null  object 
 5   start_station_latitude   183416 non-null  float64
 6   start_station_longitude  183416 non-null  float64
 7   end_station_id           183219 non-null  float64
 8   end_station_name         183219 non-null  object 
 9   end_station_latitude     183416 non-null  float64
 10  end_station_longitude    183416 non-null  float64
 11  bike_id                  183416 non-null  int64  
 12  user_type                183416 non-null  object 
 13  member_birth_year        175151 non-null  float64
 14  memb

In [6]:
## I display 15 rows to understand the content values more
df.head(15)

duration_sec start_time end_time  start_station_id  \
0          52185    32:10.1  01:56.0              21.0   
1          42521    53:21.8  42:03.1              23.0   
2          61854    13:13.2  24:08.1              86.0   
3          36490    54:26.0  02:36.8             375.0   
4           1585    54:18.5  20:44.1               7.0   
5           1793    49:58.6  19:51.8              93.0   
6           1147    55:35.1  14:42.6             300.0   
7           1615    41:06.8  08:02.8              10.0   
8           1570    41:48.8  07:59.7              10.0   
9           1049    49:47.7  07:17.0              19.0   
10           458    57:57.2  05:35.4             370.0   
11           506    56:55.5  05:21.7              44.0   
12          1176    45:12.7  04:49.2             127.0   
13           915    49:06.1  04:21.9             252.0   
14           395    56:26.8  03:01.9             243.0   

                                   start_station_name  start_station_latitude  \
0    Montgomery St BART Station (Market St at 2nd St)               37.789625   
1                       The Embarcadero at Steuart St               37.791464   
2                             Market St at Dolores St               37.769305   
3                             Grove St at Masonic Ave               37.774836   
4                                 Frank H Ogawa Plaza               37.804562   
5                        4th St at Mission Bay Blvd S               37.770407   
6                                Palm St at Willow St               37.317298   
7                          Washington St at Kearny St               37.795393   
8                          Washington St at Kearny St               37.795393   
9                                Post St at Kearny St               37.788975   
10                                Jones St at Post St               37.787327   
11  Civic Center/UN Plaza BART Station (Market St ...               37.781074   
12                             Valencia St at 21st St               37.756708   
13                       Channing Way at Shattuck Ave               37.865847   
14                        Bancroft Way at College Ave               37.869360   

    start_station_longitude  end_station_id  \
0               -122.400811            13.0   
1               -122.391034            81.0   
2               -122.426826             3.0   
3               -122.446546            70.0   
4               -122.271738           222.0   
5               -122.391198           323.0   
6               -121.884995           312.0   
7               -122.404770           127.0   
8               -122.404770           127.0   
9               -122.403452           121.0   
10              -122.413278            43.0   
11              -122.411738           343.0   
12              -122.421025           323.0   
13              -122.267443           244.0   
14              -122.254337           252.0   

                                     end_station_name  end_station_latitude  \
0                      Commercial St at Montgomery St             37.794231   
1                                  Berry St at 4th St             37.775880   
2        Powell St BART Station (Market St at 4th St)             37.786375   
3                              Central Ave at Fell St             37.773311   
4                               10th Ave at E 15th St             37.792714   
5                                  Broadway at Kearny             37.798014   
6                            San Jose Diridon Station             37.329732   
7                              Valencia St at 21st St             37.756708   
8                              Valencia St at 21st St             37.756708   
9                                  Mission Playground             37.759210   
10  San Francisco Public Library (Grove St at Hyde...             37.778768   
11                                Bryant St at 2nd St             37.783172  

In [7]:
# descriptive statistics for numeric variables
df.describe()

duration_sec  start_station_id  start_station_latitude  \
count  183416.000000     183219.000000           183416.000000   
mean      726.067311        138.590365               37.771221   
std      1794.371831        111.778938                0.099586   
min        61.000000          3.000000               37.317298   
25%       325.000000         47.000000               37.770083   
50%       514.000000        104.000000               37.780760   
75%       796.000000        239.000000               37.797280   
max     85444.000000        398.000000               37.880222   

       start_station_longitude  end_station_id  end_station_latitude  \
count            183416.000000   183219.000000         183416.000000   
mean               -122.352661      136.249401             37.771425   
std                   0.117100      111.514916              0.099494   
min                -122.453705        3.000000             37.317298   
25%                -122.412408       44.000000             37.770407   
50%                -122.398285      100.000000             37.781010   
75%                -122.286533      235.000000             37.797320   
max                -121.874119      398.000000             37.880222   

       end_station_longitude        bike_id  member_birth_year  
count          183416.000000  183416.000000      175151.000000  
mean             -122.352247    4472.890260        1984.806504  
std                 0.116677    1664.386068          10.116589  
min              -122.453705      11.000000        1878.000000  
25%              -122.411726    3777.000000        1980.000000  
50%              -122.398279    4958.000000        1987.000000  
75%              -122.288045    5502.000000        1992.000000  
max              -121.874119    6645.000000        2001.000000



> ### What is the structure of your dataset?





- The structure of the dataset contains  15  columns and 183416  rows.
 
 
- Most variables are numbers in nature , and here are **dtypes summary :**

        - float(7)
        - int(2)
        - object(7)
        



> ### What are the main interesting features in the dataset?
duration_sec
member_birth_year
member_gender
user_type
start and end station
bike_share_for_all_trip

 **We got three main feature areas which are :**
 
 
 
 
- duration_sec


 
- bike_share_for_all_trip
  
 
  
- member_birth_year




> ### What features in the dataset supported my investigation for these interesting features?



- 



- 



- 



- 




- 




##  More Exploring ideas to begin my investigation

In [8]:
#This code for describing statistical data ideas in a more organized and readable table

odesc = df.describe().T

odesc

count         mean          std          min  \
duration_sec             183416.0   726.067311  1794.371831    61.000000   
start_station_id         183219.0   138.590365   111.778938     3.000000   
start_station_latitude   183416.0    37.771221     0.099586    37.317298   
start_station_longitude  183416.0  -122.352661     0.117100  -122.453705   
end_station_id           183219.0   136.249401   111.514916     3.000000   
end_station_latitude     183416.0    37.771425     0.099494    37.317298   
end_station_longitude    183416.0  -122.352247     0.116677  -122.453705   
bike_id                  183416.0  4472.890260  1664.386068    11.000000   
member_birth_year        175151.0  1984.806504    10.116589  1878.000000   

                                 25%          50%          75%           max  
duration_sec              325.000000   514.000000   796.000000  85444.000000  
start_station_id           47.000000   104.000000   239.000000    398.000000  
start_station_latitude     37.770083    37.780760    37.797280     37.880222  
start_station_longitude  -122.412408  -122.398285  -122.286533   -121.874119  
end_station_id             44.000000   100.000000   235.000000    398.000000  
end_station_latitude       37.770407    37.781010    37.797320     37.880222  
end_station_longitude    -122.411726  -122.398279  -122.288045   -121.874119  
bike_id                  3777.000000  4958.000000  5502.000000   6645.000000  
member_birth_year        1980.000000  1987.000000  1992.000000   2001.000000

In [9]:
# display the information about each column
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 183416 entries, 0 to 183415
Data columns (total 16 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   duration_sec             183416 non-null  int64  
 1   start_time               183416 non-null  object 
 2   end_time                 183416 non-null  object 
 3   start_station_id         183219 non-null  float64
 4   start_station_name       183219 non-null  object 
 5   start_station_latitude   183416 non-null  float64
 6   start_station_longitude  183416 non-null  float64
 7   end_station_id           183219 non-null  float64
 8   end_station_name         183219 non-null  object 
 9   end_station_latitude     183416 non-null  float64
 10  end_station_longitude    183416 non-null  float64
 11  bike_id                  183416 non-null  int64  
 12  user_type                183416 non-null  object 
 13  member_birth_year        175151 non-null  float64
 14  memb

## <mark>The above result show some missing values exist in some columns So let's display them </mark>

In [10]:
#YOUR CODES
  missing_values = df.isnull().sum()
missing_values_percentage = (missing_values / len(df)) * 100

print("\nMissing Values in Each Column:\n")
     print(missing_values)
print("\nPercentage of Missing Values in Each Column:\n")
print(missing_values_percentage)

IndentationError: unexpected indent (506278669.py, line 2)



> ## <mark>The result above shows many missing data </mark>



###  To fix this issue You will fill these values with mean for numeric and mod for categorical

In [ ]:
# trying to fill missing values in my datset using for loop
# we will delete the missing values for start_station_id , start_station_name , end_station_id , end_station_name
columns_to_fill = ['start_station_id', 'start_station_name', 'end_station_id', 'end_station_name']
for column in columns_to_fill:
    df[column].fillna('Unknown', inplace=True)

# for member_birth_year we will fill with mean value
df['member_birth_year'].fillna(df['member_birth_year'].mean(), inplace=True)


In [ ]:
#check the Nulls
df.isnull().sum()

### **And with a proper category for other (member gender)**

In [ ]:
##  still got a missing value in the member_gender, so let's explore this column more 
gbf = df.copy()
print(gbf.member_gender.head(15))

In [ ]:
# fill  dataset with missing values.
gbf['member_gender'].fillna(gbf['member_gender'].mode()[0], inplace=True)

In [ ]:
print(gbf.member_gender.head(15))

In [ ]:
# delete all other in the member_gender column
gbf = gbf[gbf['member_gender'] != 'Other']
## disply unique values in member_gender
gbf['member_gender'].unique()

In [ ]:
## check my result above
gbf.isnull().sum()

> ## <mark>The result above shows that we deal with all missing data </mark>

### Now let's check duplicate value

In [ ]:
## check duplicate in dataset
df = gbf.copy()
duplicate_rows = df.duplicated()
print("Number of duplicate rows: ", duplicate_rows.sum())

> ##If Not significant duplicated , so won't clean it 


### Now let's check the category columns to make more cleaning dataset 

In [ ]:
### check the category columns that included object in a table view usig .T
cat_obj = df.describe(include=['object']).T
cat_obj

 > ## From all previous results above I found  :
 
 
 
 **=========================================================**
 
 

> 1 ) **unique Value** as in ((`member_gender`)) : Not significant , so won't clean it



> 2 ) **Incorrect data type** as in (( `user_type` &&	`member_gender`)) :  significant change , so I need to clean it.


> 3 ) **Good factor** for marketing work as in ((`member_birth_year`)) : so I need to clean it.

In [ ]:
##  step  2 ) Incorrect data type as in (( `user_type` &&	`member_gender`)) :  significant change , so  need to clean it.
df['user_type'] = df['user_type'].astype('category')
df['member_gender'] = df['member_gender'].astype('category')

In [ ]:
## check your work
df.info()

In [ ]:
## step 3 ) Good factor for marketing work as in ((`member_birth_year`)) : so need to clean it.
## To know the target audience's age and serve on them

# create a column for users' age depending on the current year
current_year = 2024
df['member_age'] = current_year - df['member_birth_year']
# check the new column
df[['member_birth_year', 'member_age']].head(10)
df[['member_birth_year','member_age']]

In [ ]:
## check your work
df['member_age'].head(10)

In [ ]:
## change the new column age data type from float to int
# round the age values to the nearest whole number
df['member_age'] = df['member_age'].round().astype('int')
## check your work
df['member_age'].head(10)

In [ ]:
## create a function to know the average users ages
def average_user_age(dataframe):
    return dataframe['member_age'].mean()

avg_user = average_user_age(df)
print("The average user age is:", avg_user.round(2))



In [ ]:
## another important factor to offer  marketing services is the  duration sec time 
## need to change the time from second to minute to be more usable
df['trip_duration_minutes'] = df['duration_sec'] / 60
## check your work
df['trip_duration_minutes'].head(10)

In [ ]:
## I need to change the time from second to Houre  to be more usable
df['trip_duration_hours'] = df['duration_sec'] / 3600

In [ ]:
## check your work
df['trip_duration_hours'].head(10)


# <p style="border-radius:15px;border-style:solid;border-color:lightblue;border-width:3px;background-color:lightblue;padding: 20px;font-family:Verdana;color:#D81F26;font-size:30px;text-align: center;">Univariate Exploration</p>
# <a id="011"> Univariate Exploration</a>



> In this section, investigate distributions of individual variables. If
you see unusual points or outliers, take a deeper look to clean things up
and prepare yourself to look at relationships between variables.




# <a id="1"> 1 ) . What is the distribution for User type?</a>

In [ ]:


import seaborn as sns
plt.figure(figsize = (10,5), dpi = 200)

plot = sns.countplot(data=df, x='user_type')

plt.xlabel('User distribution')
plt.ylabel('Frequency')
plt.title("Represent the user's distribution depending on being a part of the Trips Program")

plt.show()

# <a id="2"> 2 ) What is the distribution for the bike share users on all Trips?</a>

In [ ]:

plt.figure(figsize=(10, 5), dpi=200)
sb.countplot(data=df, x='bike_share_for_all_trip')
plt.xlabel('User distribution')
plt.ylabel('Frequency')
plt.title("Represent the user's distribution depending on being a part of the Trips Program")
plt.show()

# <a id="3"> 3 ) What is the distribution for users' ages? </a>

In [ ]:

plt.figure(figsize=(10, 5), dpi=200)
sb.histplot(data=df, x='member_age', bins=30, kde=True)
plt.xlabel('User Ages')
plt.ylabel('Frequency')
plt.title("Distribution of Users' Ages")
plt.show()


# <a id="4">  4 ) . Do the users' ages have outlier values?</a>


In [ ]:

plt.figure(figsize=(8, 6), dpi=200)
sb.boxplot(data=df, x='member_age')
plt.xlabel('Member Age')
plt.title('Boxplot of Member Ages')
plt.show()

# <a id="5">  5 ) . What is the distribution for the trip duration in minutes? </a>

In [ ]:
## Review the statistic for trip duration in minutes
df['trip_duration_minutes'].describe()

In [ ]:
# Make visualization to easy get insight about trip duration in minutes
plt.figure(figsize=(10, 5), dpi=200)
sb.histplot(data=df, x='trip_duration_minutes', bins=100, kde=True)
plt.xlabel('Trip Duration (Minutes)')
plt.ylabel('Frequency')
plt.title('Distribution of Trip Duration in Minutes')
plt.show()


# <a id="6"> 6 ) . What is the distribution for member_gender?</a>

In [ ]:
# Make visualization to easy get insight member gender
plt.figure(figsize=(10, 5), dpi=200)
sb.countplot(data=df, x='member_gender')
plt.xlabel('Member Gender')
plt.ylabel('Frequency')
plt.title('Distribution of Member Gender')
plt.show()

# <a id="7">    7 )  . What is the distribution for the trip duration in Hours?  </a>

In [ ]:
# Make visualization to easy get insight distribution for the trip duration in hours
plt.figure(figsize=(10, 5), dpi=200)
sb.histplot(data=df, x='trip_duration_hours', bins=50, kde=True)
plt.xlabel('Trip Duration (Hours)')
plt.ylabel('Frequency')
plt.title('Distribution of Trip Duration in Hours')
plt.show()

### The variables distribution:


> The distribution of duration present that most of the trips were less than 30 minutes ( we can notice that from the minutes and the hour charts )


> And the Age distribution is ( left Skewed ) as most users age are  22 - 39


> Most users are subscribers, not Customers and  Males as shown in the charts above.



### Were there any interesting and outliers 


- We got Age outliers for for more than 100 .



- The distribution of duration was strange where the minutes and the hour charts were 30 minutes or less.

# <p style="border-radius:15px;border-style:solid;border-color:lightblue;border-width:3px;background-color:lightblue;padding: 20px;font-family:Verdana;color:#D81F26;font-size:30px;text-align: center;">Bivariate Exploration</p>
# <a id="111"> Bivariate Exploration</a>

Now let's investigate relationships between pairs of variables in our data.

# <a id="8"> 8) . What the average duration of trip for the categories of user type?</a>

In [ ]:
# the average duration of trip for the categories of user type
avg_duration_by_user_type = df.groupby('user_type')['trip_duration_minutes'].mean().reset_index()
plt.figure(figsize=(8, 6), dpi=200)
sb.barplot(data=avg_duration_by_user_type, x='user_type', y='trip_duration_minutes')
plt.xlabel('User Type')
plt.ylabel('Average Trip Duration (Minutes)')
plt.title('Average Trip Duration by User Type')
plt.show()

# Make visualization to easy get insight


# <a id="9">    9 ) . How long is the duration of the trip for each gender in minutes?  </a>

In [ ]:
# duration of the trip for each gender in minutes scatter plot
plt.figure(figsize=(10, 5), dpi=200)
sb.scatterplot(data=df, x='member_gender', y='trip_duration_minutes', alpha=0.5)
plt.xlabel('Member Age')
plt.ylabel('Trip Duration (Minutes)')
plt.title('Trip Duration vs. Member Age by Gender')
plt.show()





# plt.figure(figsize=(10, 5), dpi=200)
# sb.boxplot(data=df, x='member_gender', y='trip_duration_minutes')
# plt.xlabel('Member Gender')
# plt.ylabel('Trip Duration (Minutes)')
# plt.title('Duration of Trip by Member Gender')
# plt.show()

# Make visualization to easy get insight


# <a id="10 "> 10 ) . How long is the duration of the trip for each user_type in minutes?  </a>

In [ ]:

# Make visualization to easy get insight
# How long is the duration of the trip for each user_type in minutes
plt.figure(figsize=(10, 5), dpi=200)
sb.boxplot(data=df, x='user_type', y='trip_duration_minutes')
plt.xlabel('User Type')
plt.ylabel('Trip Duration (Minutes)')
plt.title('Duration of Trip by User Type')
plt.show()

> ### <mark>Your analsys and exploration here </mark>

# <a id="11">  11 ) . How the user distribution depending on their Age and user_type ? </a>

In [ ]:

# Make visualization to easy get insight
# How the user distribution depending on their Age and user_type
plt.figure(figsize=(10, 5), dpi=200)
sb.boxplot(data=df, x='user_type', y='member_age')
plt.xlabel('User Type')
plt.ylabel('Member Age')
plt.title('Distribution of Member Age by User Type')
plt.show()


> ### <mark>Your analsys and exploration here </mark>

#  <a id="12"> 12 ). How the user distribution depending on their Age and member_gender ?</a>

In [ ]:
# Make visualization to easy get insight


> ### <mark>Your analsys and exploration here </mark>

### Talk about some of the relationships you observed in this part of the investigation. How did the feature(s) of interest vary with other features in the dataset?

> ...................


### Did you observe any interesting relationships between the other features (not the main feature(s) of interest)?

> ....................

# <p style="border-radius:15px;border-style:solid;border-color:lightblue;border-width:3px;background-color:lightblue;padding: 20px;font-family:Verdana;color:#D81F26;font-size:30px;text-align: center;">Multivariate Exploration</p>
# <a id="1111"> Multivariate Exploration </a>

> Create plots of three or more variables to investigate your data even
further. Make sure that your investigations are justified, and follow from
your work in the previous sections.

In [ ]:
## check your work 



# <a id="13">  13 ) . What is the  Correlation between each variable in the dataset?</a>

In [ ]:
# Make visualization to easy get insight




> ### <mark>Your analsys and exploration here </mark>

Example usage
- There is `no Correlation` for any variable with time ( `duration sec /minutes/hours`)



# <a id="14"> 14 )  .  What is the  Correlation between `specific numeric variables` in the dataset?</a>


In [ ]:
# Make visualization to easy get insight




> ### <mark>Your analsys and exploration here </mark>

Example usage
- There is `no Correlation` for any variable with time ( `duration sec /minutes`)




- 




- 




# <a id="15">  15 ) What is the  Correlation between data_stat variables in the dataset?</a>


In [ ]:

"""""This time I will use heatmap  to explain how Correlation between data_stat variables

in the dataset in more clear and specific percentage"""

# Make visualization to easy get insight



> ### <mark>The above charts show how Correlation between specific numeric variables in the dataset using heatmap plots  </mark>



**Actually the `heat map`  display the above ideas in clear percentages**



- There is `no Correlation` for any variable with time ( `duration sec /minutes`)




- There is a `negative  Correlation` between  member_birth_year  and age which normal.




- There is `no significant Correlation` for any variable with `bike_id `



# <p style="border-radius:15px;border-style:solid;border-color:lightblue;border-width:3px;background-color:lightblue;padding: 20px;font-family:Verdana;color:#D81F26;font-size:30px;text-align: center;">Conclusions</p>
# <a id="1100"> Conclusions </a>

#example
- Most users were Male.


- 

- 

- 



## limitation 
example
- We got missing data which we fix some of them .

- 
